core/domain/collection_services.py

```diff
@@ -663,14 +668,18 @@ def apply_change_list(collection_id, change_list):
                 # latest schema version. As a result, simply resaving the
                 # collection is sufficient to apply the schema migration.
                 continue
+
         return collection
 
     except Exception as e:
         logging.error(
             '%s %s %s %s' % (
                 e.__class__.__name__, e, collection_id, change_list)
         )
-        raise
+        # This code is needed in order to reraise the error properly with
+        # the stacktrace. See https://stackoverflow.com/a/18188660/3688189.
+        exec_info = sys.exc_info()
+        six.reraise(exec_info[0], exec_info[1], tb=exec_info[2])
```

Comment:
Is 'six' an interop library for Python 2/3? I'm wondering whether we can't use our own Python utils to bridge this temporarily until we're fully on Python 3 and can just do 'raise e'.

Reply:
+1, we should not expose six outside of python_utils (I've filed #11481 to have this enforced by lint checks).

Reply:
Makes sense, done.